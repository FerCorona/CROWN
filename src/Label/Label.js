import React from 'react';
import { LABEL_SIZE,LABEL_WEIGHT } from '../Common/Constants';

const Label = ({ text, color, size, weight }) => {
  const style = {
    color: `var(${color})`,
    fontSize: `${LABEL_SIZE[size]}`,
    fontWeight: `${LABEL_WEIGHT[weight]}`
  };
  return (
    <div
      className='CROWNLabel'
      style={style}
    >{text}
    </div>
  );
};

Label.defaultProps = {
  text: 'I\'m a label',
  size: 'small',
  color: '--black-0',
  weight: 'regular'
};

export default Label;