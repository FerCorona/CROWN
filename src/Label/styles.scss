.CROWNLabel {
    font-family: -apple-system, BlinkMacSystemFont, sans-serif;
    margin-top: 10px;
    margin-right: 0px;
    margin-bottom: 5px;
    margin-left: 0px;
}