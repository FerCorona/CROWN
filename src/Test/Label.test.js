import React from 'react';
import renderer from 'react-test-renderer';
import Label from '../Label/index';

test('Link changes the class when hovered', () => {
  const component = renderer.create(
    <Label />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});