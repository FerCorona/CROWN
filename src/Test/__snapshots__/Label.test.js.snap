// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<div
  className="CROWNLabel "
  style={
    Object {
      "color": "var(--black-1)",
      "fontSize": "22px",
      "fontWeight": "400",
    }
  }
>
  I'm a label
</div>
`;
